### PR TITLE
Navigation view fixes

### PR DESCRIPTION
**Pull request by danielosman**
Fixed empty navigation view update and navigation selection growing due to rounding errors.

### CODE REVIEW

File: src/composites/navigation/NavigationView.js [hunk 1]
```diff
@@ -137,14 +158,15 @@ export default class NavigationView extends ChartView {
       this._ticking = true
     }
   }
+
```
Comment (Dmitra) [inline src/composites/navigation/NavigationView.js]: I remember we agreed to not put empty line between functions already splitted by multiline comment

File: src/composites/navigation/NavigationView.js [hunk 2]
```diff
@@ -117,17 +124,31 @@ export default class NavigationView extends ChartView {
...
+  */
   _onSelection (range) {
     const xAccessor = this.config.get('plot.x.accessor')
     const xScale = this._yChart.config.get('axes.x.scale')
     let xMin = xScale.invert(range[0])
     let xMax = xScale.invert(range[1])
-    const sScale = this.config.get('selectionScale')
-    this.config.set('selection', [sScale.invert(range[0]), sScale.invert(range[1])], { silent: true })
-    this.config.set('pixelSelection', range, { silent: true })
+    if (!this._updatedFromNavigationView) {
```
Comment (Dmitra) [inline src/composites/navigation/NavigationView.js]: please, add comments in the code on why do we need to differentiate on source of update

File: src/composites/navigation/NavigationView.js [hunk 3]
```diff
@@ -117,17 +124,31 @@ export default class NavigationView extends ChartView {
...
+      const sScale = this.config.get('selectionScale')
+      this.config.set('selection', [sScale.invert(range[0]), sScale.invert(range[1])], { silent: true })
+      this.config.set('pixelSelection', range, { silent: true })
```
Comment (Dmitra) [inline src/composites/navigation/NavigationView.js]: please, comment in the code on what is pixelSelection and why do another selection

File: src/composites/navigation/NavigationView.js (hunk 1 above)
Comment (danielosman) [reply src/composites/navigation/NavigationView.js]: Don't recall that, I thought lint tells us what is acceptable.

File: src/composites/navigation/NavigationView.js (hunk 2 above)
Comment (danielosman) [reply src/composites/navigation/NavigationView.js]: I will review, perhaps not needed any more.

**Pull request merged**